### PR TITLE
1130: validation of service indicators

### PR DESCRIPTION
**Pull request by rachaelparris**

### CODE REVIEW

File: shared/src/proxies/manualAssociation/associateRespondentWithCaseProxy.js [hunk 1]
```diff
@@ -12,11 +12,12 @@ const { post } = require('../requests');
 exports.associateRespondentWithCaseInteractor = ({
   applicationContext,
   caseId,
+  serviceIndicator,
```
Comment (kkoskelin) [inline shared/src/proxies/manualAssociation/associateRespondentWithCaseProxy.js:16]: will need jsdoc update

File: web-client/src/presenter/actions/ManualAssociation/associateRespondentWithCaseAction.test.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import { associateRespondentWithCaseAction } from './associateRespondentWithCaseAction';
+import { constants } from '../../../../../shared/src/business/utilities/setServiceIndicatorsForCase';
```
Comment (kkoskelin) [inline web-client/src/presenter/actions/ManualAssociation/associateRespondentWithCaseAction.test.js:2]: Should we be instead acquiring these constants from the applicationContext?

**Review by codyseibert — APPROVED** [review]

File: web-client/src/presenter/actions/validatePetitionerInformationFormAction.js [hunk 3]
```diff
@@ -26,6 +31,35 @@ export const validatePetitionerInformationFormAction = ({
       partyType,
     });
 
+  if (
+    ['Paper', 'None'].includes(oldContactPrimary.serviceIndicator) &&
+    contactPrimary.serviceIndicator === 'Electronic'
+  ) {
+    const serviceIndicatorError = {
```
Comment (kkoskelin) [inline web-client/src/presenter/actions/validatePetitionerInformationFormAction.js]: maybe we could DRY this up a little and move the declaration of `serviceIndicatorError` before the first of the two `if` blocks?

File: web-client/src/views/CaseDetail/EditPractitionersModal.jsx [hunk 4]
```diff
@@ -119,7 +120,12 @@ export const EditPractitionersModal = connect(
                   <div className="margin-top-2">
                     <ServiceIndicatorRadios
                       bind={`modal.practitioners.${idx}`}
-                      validationErrors="validationErrors"
+                      getValidationError={() =>
+                        validationErrors.practitioners &&
+                        validationErrors.practitioners[idx] &&
+                        validationErrors.practitioners[idx].serviceIndicator
```
Comment (kkoskelin) [inline web-client/src/views/CaseDetail/EditPractitionersModal.jsx:126]: If you wanted to go *really* crazy (via Babel 7.8.0ish) you could
`validationErrors.practitioners?[idx]?.serviceIndicator`

**Review by kkoskelin — CHANGES_REQUESTED** [review]
Probably need to review jsdocs for some of these functions (I saw two that needed updates and stopped looking).  Also I had a question about whether we should be pulling constants via `require` to a shared folder vs using applicationContext.  And finally just some nitpicks you can ignore about coding style -- I saw several opportunities to use the `existential?.operator?()` within JSX components.  (I believe it works in JSX but possibly not in actions, fwiw)

File: web-client/src/views/CaseDetail/EditPractitionersModal.jsx (hunk 4 above)
Comment (rachaelparris) [reply web-client/src/views/CaseDetail/EditPractitionersModal.jsx:126]: It's not working with our current lint rules

**Review by kkoskelin — APPROVED** [review]

**Pull request merged**